pennylane/qnode.py

```diff
@@ -172,6 +172,8 @@ class QNode:
             the maximum number of derivatives to support. Increasing this value allows
             for higher order derivatives to be extracted, at the cost of additional
             (classical) computational overhead during the backwards pass.
+        device_vjp=False (bool): Whether or not to use the device provided Vector Jacobian Product (VJP).
+            A value of ``None`` indicates to use it if the device provides it, but use the full jacobian otherwise.

```

Comment:
Similar behaviour to `grad_on_execution`. `None` means "you can make the best choice".

Reply:
> Similar behaviour to `grad_on_execution`. `None` means "you can make the best choice".

If this is the case why is the default not `None`?

Reply:
> Similar behaviour to `grad_on_execution`. `None` means "you can make the best choice".

If this is the case then doesn't it make more sense for the default to be `None` instead of `False`?

Reply:
Right now the best choice is rarely device vjps.  

PR #4841  will solve some of the performance issues, and make device vjps substantially faster than full jacobians for the gradient case, but `device_vjp` is still quite a bit slower for the full jacobian case.

We also don't support all interfaces, and even when we do add jax (non-jit) support in #4878, it will only support `jax.grad` and `jax.vjp`, not `jax.jacobian`.